actions/python-setup-poetry/action.yaml

```diff
@@ -43,23 +48,18 @@ runs:
 
     - name: Install Poetry ${{ inputs.poetry-version }}
       if: steps.cache-poetry.outputs.cache-hit != 'true'
-      run: pipx install poetry==${{ inputs.poetry-version }}
       shell: bash
+      run: pipx install poetry==${{ inputs.poetry-version }} --python "${{ steps.setup-python.outputs.python-path }}"
 
-    - name: Configure Poetry
-      run: |
-        poetry config virtualenvs.in-project true
-        poetry config virtualenvs.path ${{ inputs.working-directory }}/.venv
-      shell: bash
-
-    - name: Set up Python ${{ inputs.python-version }}
-      uses: actions/setup-python@v4
-      id: setup-python
+    - name: Set up Python ${{ inputs.python-version }} poetry cache
+      uses: actions/setup-python@v5
```

Comment:
Have you tried to merely pull up this `setup-python` step to the top instead of introducing a 2nd one at the top? Perhaps that's already enough to fix this.

Reply:
I tried that. It failed requiring the poetry binary.

Reply:
How do the `setup-python` steps relate to Poetry being available? That happens through `pipx install`. If that was really the issue, it sounds a bit like an issue with GHA cache?

Reply:
https://github.com/bakdata/github-playground-2/actions/runs/16801695033/job/47584394585
this is the correct run that points out that the binary is misisng

Reply:
Yes I get it now that it's due to the poetry dependency caching that's part of `setup-python`